Rerun transaction on error; remove Circuits.I2C retries

doawoo opened the pull request:
* Allows the read operation to be retried if the CRC value returning is invalid
* Uses the same retry count as the I2C built-in retry count

Commit:
Retry the whole transaction when there's an error

From reading the datasheet, it really doesn't look like issuing a second
read if there's an error on the first one is guaranteed to work. It
looks safer to reissue the whole thing by sending the commend and
reading 6-bytes again.

This means that using Circuits.I2C to retry writes or reads might not do
the right thing. It looks safer to retry the transaction as a whole.

fhunleth commented:
Rebased and updated to only retry the read on a CRC error. The previous way would re-issue the whole request which was a little more than needed. This still needs a test which will be coming on the next pass - probably also a little code cleanup.

fhunleth commented:
Upon rereading the datasheet, I changed this around to completely reissue the transaction whenever there's a CRC error. Using Circuit.I2C's retries seems risky. This might be what causes intermittent glitches on SHT41 reads, but that will take more testing. A circuits_sim update is needed for the tests to pass.

fhunleth committed:
Reduce retries to 2, so only trying 3 times total

There was a misconception that I2C requests were tried 3 times rather
than retried 3 times (i.e., tried 4 times total). This change matches
the expection.
